test/infrastructure/inmemory/internal/controllers/inmemorymachine_controller.go

```diff
@@ -1012,6 +1030,8 @@ func (r *InMemoryMachineReconciler) reconcileDeleteNode(ctx context.Context, clu
 			Name: inMemoryMachine.Name,
 		},
 	}
+
+	// This is part of the MachineController's flow so we shouldn't manage it here.
```

Comment:
q: is this something we are going to fix in this PR? Otherwise let's turn this into
```suggestion
	// TODO: check if we can drop this given that the MachineController is already draining pods and deleting nodes.
```
Note to self: we are using pods, kubeadm uses static pods; TBD if this is something else to be fixed, but for sure not in this PR